app/controllers/concerns/yaml_helpers.rb

```diff
@@ -0,0 +1,35 @@
+module YamlHelpers
+  def load_yaml(file_path, values=nil)
+    data = YAML.load(File.open("#{Rails.root}/#{file_path}", 'r'))
+
+    if values.present? then 
```

Comment:
The keyword `then` can be omitted.
Also, could you please clarify what's the purpose of this function here?
I can see you are using it alongside the stats serializer, but couldn't understand what's the idea behind this.

Reply:
The idea is that variables can be inserted into yaml content. A hash is passed with keys that match up with `${hash_key}` in the yaml strings. The value of hash[key] is then used to replace the bit in ${} in the yaml string. For consistency I've then used this load_yaml method to load all of the yaml strings, even if no hash is passed with variables. Does that make sense?

Reply:
This is used in chapter two to add all the stats into the chapter text. So check out chapter-2.yml and the chapter_2 action in the chapters controller to see it being used. Hopefully that helps explain what's going on!

Reply:
Cool, I think I got it now, thanks!
Could you please add some comments that roughly explain the behaviour please?
Also, is there a specific reason for the `begin rescue` block? It looks like all possible errors are being handled manually already

Reply:
I think I did this because otherwise the error raised just kills the page and that seemed a bit of a dramatic response, possibly if the issue occurred on production, but then again maybe it is appropriate to do that, so you know there is an issue straight away and can fix it. I wasn't really sure how to handle the error.

If the question is actually why I'm raising the error in the first place - this was just to make the error easier to spot and fix.

Reply:
Is there any error you are expecting it to raise?
To me it looks like the only other error might be happening when `hash` is nil.
So perhaps you could handle that manually as well.

Otherwise, I would raise the error to AppSignal as well (if we have it set up) within the rescue block

Reply:
We don't have appsignal setup for pp live report.

The main problem will just be mismatching of key names and the names in the yaml strings, so I want to check that there aren't any yaml placeholders that don't have a corresponding value in the hash. Checking for the hash as nil is also a good idea.